docs/en/latest/plugins/public-api.md

```diff
@@ -0,0 +1,132 @@
+
+With the `public-api` plugin, we put the all public API into general HTTP API router, which is consistent with the normal route registered by the user and can apply any plugin. The public API added in the user plugin is no longer open by default by APISIX, but the user manually configures the route for it, and the user can configure any uri and plugin.
+
+## Attributes
+
+| Name | Type | Requirement | Default | Valid | Description |
+| -- | -- | -- | -- | -- | -- |
+| uri | string | optional | "" |   | The uri of the public API. When you set up the route, you can use this to configure the original API uri if it is used in a way that is inconsistent with the original public API uri. |
+
+## Example
+
+We take the `jwt-auth` token sign API as an example to show how to configure the `public-api` plugin. Also, the `key-auth` will be used to show how to configure the protection plugin for the public API.
+
+### Prerequisites
+
+The use of key-auth and jwt-auth requires the configuration of a consumer that contains the configuration of these plugins, and you need to create one in advance, the process will be omitted here.
+
+### Basic Use Case
+
+First we will setup a route.
+
+```shell
+$ curl -X PUT 'http://127.0.0.1:9080/apisix/admin/routes/r1' \
+    -H 'X-API-KEY: <api-key>' \
+    -H 'Content-Type: application/json' \
+    -d '{
+    "uri": "/apisix/plugin/jwt/sign",
+    "plugins": {
+        "public-api": {}
+    }
+}'
+```
+
+Let's test it.
+
+```shell
+$ curl 'http://127.0.0.1:9080/apisix/plugin/jwt/sign?key=user-key'
+```
+
+It will respond to a text JWT.
+
+### Customize URI
+
+Let's setup another route.
+
+```shell
+$ curl -X PUT 'http://127.0.0.1:9080/apisix/admin/routes/r2' \
+    -H 'X-API-KEY: <api-key>' \
+    -H 'Content-Type: application/json' \
+    -d '{
+    "uri": "/gen_token",
+    "plugins": {
+        "public-api": {
+            "uri": "/apisix/plugin/jwt/sign"
+        }
+    }
+}'
+```
+
+Let's test it.
+
+```shell
+$ curl 'http://127.0.0.1:9080/gen_token?key=user-key'
+```
+
+It will still respond to a text JWT. We can see that users are free to configure URI for the public API to match.
+
+### Protect Route
+
+Let's modify the last route and add `key-auth` authentication to it.
+
+```shell
+$ curl -X PUT 'http://127.0.0.1:9080/apisix/admin/routes/r2' \
+    -H 'X-API-KEY: <api-key>' \
+    -H 'Content-Type: application/json' \
+    -d '{
+    "uri": "/gen_token",
+    "plugins": {
+        "public-api": {
+            "uri": "/apisix/plugin/jwt/sign"
+        },
+        "key-auth": {}
+    }
+}'
+```
+
+Let's test it.
+
+```shell
+$ curl -i 'http://127.0.0.1:9080/gen_token?key=user-key'
+    -H "apikey: test-apikey"
+HTTP/1.1 200 OK
+
+# Failed request
+$ curl -i 'http://127.0.0.1:9080/gen_token?key=user-key'
+HTTP/1.1 401 UNAUTHORIZED
+```
+
```

Comment:
how about use 

```
$ curl 'http://127.0.0.1:9080/apisix/plugin/jwt/sign?key=user-key'
```

to get token atfer protecting this route?

Reply:
This is inherited from the second case. If you don't use the `/get_token` route, you can just configure the public API route in the first way (Basic Use Case).

Reply:
Does this mean that the `/apisix/plugin/jwt/sign` is still exposed?

Reply:
@tzssangglass 

It will still be inserted into the radixtree built in `api_router` at startup, but in the default `http_access_phase`, no match will be made. If the user configures a route for it and turns on the `public-api` plugin, we do an `api_router` match in the `public-api` plugin's access hook to process the public API request.

Therefore, the conclusion is that it is not exposed and the user needs to configure the route to access it.

Reply:
got